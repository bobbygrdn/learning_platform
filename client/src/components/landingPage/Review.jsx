import React from 'react';
import '../../styles/landingPage/Review.css';

export default function Review({ review }) {
    return (
        <>
            <q className='reviewText' data-testid="reviewText">{review.quote}</q>
            <p className='user' data-testid="user">{review.user}</p>
        </>
    )
}
