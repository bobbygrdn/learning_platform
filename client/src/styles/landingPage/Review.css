.reviewText {
    text-align: left;
}

.user {
    text-align: left;
    padding-right: 2em;
    margin-bottom: 1em;
    font-weight: bold;
}