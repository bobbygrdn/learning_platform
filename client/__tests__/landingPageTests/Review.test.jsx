import React from 'react';
import { render, screen } from '@testing-library/react';
import Review from '../../src/components/landingPage/Review';

const review = () => {
    return {
        quote: 'This is a quote',
        user: 'This is a user',
    }
}

it('renders without crashing', () => {
    const { container } = render(<Review review={review()} />);
    expect(container).toMatchSnapshot();
})

it('should render quote element with text class name', () => {
    render(<Review review={review()} />);
    expect(screen.getAllByTestId('reviewText')).toHaveLength(1);
})

it('should render paragraph element with user class name', () => {
    render(<Review review={review()} />);
    expect(screen.getAllByTestId('user')).toHaveLength(1);
})