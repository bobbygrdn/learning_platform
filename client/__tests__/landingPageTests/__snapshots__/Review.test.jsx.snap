// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <q
    class="reviewText"
    data-testid="reviewText"
  >
    This is a quote
  </q>
  <p
    class="user"
    data-testid="user"
  >
    This is a user
  </p>
</div>
`;
